cmd/pebble/test.go

```diff
@@ -329,6 +329,17 @@ func runTest(dir string, t test) {
 		close(workersDone)
 	}()
 
+	if maxSize > 0 {
+		go func() {
+			for {
+				time.Sleep(10 * time.Second)
+				if db.Metrics().DiskSpaceUsage() > maxSize*1e6 {
+					fmt.Println("max size reached")
+					done <- syscall.Signal(0)
```

Comment:
nit: can return after this to not leave a dangling goroutine. Maybe this file needs an update so that we actually terminate both this goroutine and the one below it when this method finishes, as currently that's not guaranteed. But that can wait for a TODO.

Reply:
general go question: given that this syscall.Signal() method will tear down the go process, which cancels all go routines, what's the point of avoiding a dangling go routine here? For a more graceful shutdown?

Reply:
In this case it's okay as long as all callers of `runTest` terminate immediately after it. It's just a bad idea from a general hygiene standpoint because `runTest` is the one leaking a goroutine permanently even after it finishes, and that can be a cpu/memory issue if it's called repeatedly or the process sticks around after the function call.

Also, `syscall.Signal` isn't doing anything here, it's just a type to hold a signal value. If we got a SIGABRT that'd also go into the `done` channel thanks to the `syscall.Notify` above. We read from the `done` channel and terminate the below loop when we see something in it.